kilkari/src/main/java/org/motechproject/nms/kilkari/service/impl/MctsBeneficiaryImportServiceImpl.java

```diff
@@ -419,13 +424,17 @@ public ChildImportRejection importChildRecord(Map<String, Object> record, Subscr
             return createUpdateChildRejections(flagForMcts, record, action, RejectionReasons.UPDATED_RECORD_ALREADY_EXISTS, false);
         }
 
-        List<DeactivatedBeneficiary> deactivatedUsers = deactivatedBeneficiaryDataService.findByExternalId(childId);
-        if (deactivatedUsers != null && deactivatedUsers.size() > 0) {
-            for (DeactivatedBeneficiary deactivatedUser : deactivatedUsers) {
-                if (deactivatedUser.getOrigin() == importOrigin) {
-                    String message = deactivatedUser.isCompletedSubscription() ? "Subscription completed" : "User deactivated";
-                    if (message.length() > 2) {
-                        return createUpdateChildRejections(flagForMcts, record, action, RejectionReasons.UPDATED_RECORD_ALREADY_EXISTS, false);
+        if (childById != null && subscriptionService.getActiveSubscription(subscriberService.getSubscriberByBeneficiary(childById), SubscriptionPackType.PREGNANCY) != null && subscriptionService.getActiveSubscription(subscriberService.getSubscriberByBeneficiary(childById), SubscriptionPackType.PREGNANCY).getStatus() == SubscriptionStatus.ACTIVE) {
+            LOGGER.debug("Active mother. We ignore the deactivated case scenario.");
+        } else {
+            List<DeactivatedBeneficiary> deactivatedUsers = deactivatedBeneficiaryService.findByExternalId(childId);
+            if (deactivatedUsers != null && deactivatedUsers.size() > 0) {
+                for (DeactivatedBeneficiary deactivatedUser : deactivatedUsers) {
+                    if (deactivatedUser.getOrigin() == importOrigin) {
+                        String message = deactivatedUser.isCompletedSubscription() ? "Subscription completed" : "User deactivated";
+                        if (message.length() > 2) {
```

Comment:
based on the above assignment to message, it seems message length will always be greater than 2 ??